### ISSUE TITLE
Setup authentication

### ISSUE THREAD

**Issue by iamareebjamal** [fossasia/open-event-next#33]
Suggest which auth library to use. We don't want server side auth, only client side

The auth library should provide a way to programmatically set logged in user. This is to ensure that we can set authenticated user across ember and react site. It should be able to persist authenticated user across both sites so user can visit them seamlessly

**Comment by iamdevvalecha** [comment]
can we use Authentication for API Routes using JWT and bcrypt?

**Comment by iamareebjamal** [comment]
See how it's done in ember project

**Comment by pc-beast** [comment]
I am working on this

**Comment by pc-beast** [comment]
i think next-auth can be used for out purposes https://next-auth.js.org/configuration/providers#using-a-custom-provider
we can use existing flask server as provider.

**Comment by iamareebjamal** [comment]
Next auth makes server part required. If it can be used to make client only auth, then go ahead

**Comment by pc-beast** [comment]
yes, you are right about server part. I am looking for other options

**Comment by harshalkaigaonkar** [comment]
is it open?

**Comment by pc-beast** [comment]
@harshalkaigaonkar yes

**Comment by harshalkaigaonkar** [comment]
We can use [localStoragedb](https://www.npmjs.com/package/localstoragedb) for data saving on the front-end and for Authentication and Authorisation, we can use JWT or [passport-local](https://www.npmjs.com/package/passport-local).

**Comment by pc-beast** [comment]
We need the solution to be consistent with the old ember site and without using the server part of next. I don't think its possible with the solutions you proposed. But if it's possible then you can start working.

**Comment by harshalkaigaonkar** [comment]
> We need the solution to be consistent with the old ember site and without using the server part of next. I don't think its possible with the solutions you proposed. But if it's possible then you can start working.

I'm not getting what consistent here means?
Like, do we need an auth solution that authenticates on client-side only (w/o including an DB to store data) ?

**Comment by pc-beast** [comment]
Please read the complete issue description carefully.

**Comment by harshalkaigaonkar** [comment]
Yeah.

**Comment by ath34-tech** [comment]
I previously tried nextAuth.js which may meet current requirements, It is client side authentication which use providers like github or google.
So what your view?
